geps/gep-713/index.md

```diff
@@ -1,1157 +1,743 @@
 
-Thus its status as EXPERIMENTAL DO NOT USE YET.
+With that in mind, a possible solution for the discoverability problem may involve designing tools (e.g. CLI tools/plugins), new CRDs, etc that let users ask questions in terms of the real life problems they have to deal with on a daily basis, rather than shaped by the underlying technologies used in the process. For instance, a simple Kubernetes object that is used to declare the rules to process a HTTP request cannot have its status reported simply as Ready/Not ready. By being a complex object composed of multiple routing rules, potentially affected by specifications declared from other objects as well, its status MUST account for that complexity and be structured in such a way that informs the owner with respect to each possible case, whether the ones induced by the internal specification declared by the object itself or its external relationships.
 
-#### Validating Admission Controller to inform users about relevant Policy
+In other words, the discoverability problem exists and must be addressed in light of the complexity associated with the topology of contexts induced by a set of hierarchically related resources. One should always have that topology in mind while asking questions regarding the behavior of a given resource, because just like a routing object (e.g. HTTPRoute) does not exist independently from its parent contexts (e.g. Gateways) or its children (e.g. Backends), any resource in focus may be just a part of a whole.
 
-Implementations MAY supply a Validating Admission Webhook that will return a
-WARNING message when an applied object is affected by some Policy.
+### Status reporting
 
-The warning message MAY include the name, namespace, apiGroup and Kind of relevant
-Policy objects.
+#### Meta resource status
 
-##### Design Considerations
+Meta resource CRDs should define a status stanza that allows for reporting the status of the meta resource with respect to each context the resource may apply.
 
-Pro:
+The basic status conditions are:
 
-- This gives object owners a very clear signal that something some Policy is
-  going to affect their object, at apply time, which helps a lot with discoverability.
+* **Accepted**: the meta resource passed both syntactic validation by the API server and semantic validation enforced by the controller, such as whether the target objects exist.
+* **Enforced**: the meta resource’s spec is guaranteed to be fully enforced, to the extent of what the controller can ensure.
+* **Partially enforced**: parts of the meta resource’s spec is guaranteed to be enforced, while other parts are known to have been superseded by other specs, to the extent of what the controller can ensure. The status should include details highlighting which parts of the meta resource are enforced and which parts have been superseded, with the references to all other related meta resources.
```

Comment:
As long as this is not a MUST then its not a problem, but this seems like it could be quite onerous to compute. For example, imagine I have a global policy and then 1000 namespaces any of which could partially conflict. Its not great to have to 'bubble up' these to the parent.

Reply:
A concrete example of this in existing Gateway API is `attachedRoutes`, which is similarly complex for implementations to compute (efficiently)